### PR TITLE
Extract Gradle logic for Java-containing subjects

### PR DESCRIPTION
**Pull request by liblit**
Previously, the top-level `build.gradle` iterated over all subprojects and applied lots of additional build-configuration logic to subprojects that contain Java source code.  Now, that Java-specific configuration logic is extracted into its own `wala-java.gradle` script plugin, which each individual subproject can load if appropriate.  This change makes the top-level `build.gradle` a bit smaller and easier to read through.

### CODE REVIEW

File: gradle-mvn-push.gradle [hunk 1]
```diff
@@ -4,6 +4,8 @@ apply plugin: 'signing'
 final isSnapshot = version.contains('SNAPSHOT')
 
 final skipTestFixtures() {
+	println "*** $project"
```
Comment (msridhar) [inline gradle-mvn-push.gradle:7]: Is this an intended change?

File: wala-java.gradle [hunk 2]
```diff
@@ -0,0 +1,106 @@
+apply plugin: 'eclipse'
+apply plugin: 'java-library'
+apply plugin: 'java-test-fixtures'
+apply plugin: 'maven-publish'
```
Comment (msridhar) [inline wala-java.gradle:4]: Will this lead us to publish more artifacts than we did before?  I didn't think we included `maven-publish` for every project with Java source before.

**Review by msridhar — COMMENTED** [review]
Couple of questions

File: gradle-mvn-push.gradle (hunk 1 above)
Comment (liblit) [reply gradle-mvn-push.gradle:7]: Oops, nope. I'll remove this shortly. Thanks for being so attentive!

File: wala-java.gradle (hunk 2 above)
Comment (liblit) [reply wala-java.gradle:4]: I think we were already including `maven-publish` for every subproject with Java source:

https://github.com/wala/WALA/blob/17064550328e82c68fb1a8464b3c26ad56c83538/build.gradle#L57-L61

Comment (msridhar) [reply wala-java.gradle:4]: Ah yes, I was confused. `gradle-mvn-push.gradle` is only included in some places

**Review by msridhar — APPROVED** [review]

**Comment by liblit** [comment]
This change worked fine on a side branch, but caused [a regression in a task that only runs on `master`](https://github.com/wala/WALA/runs/4424048891?check_suite_focus=true). I am investigating.

**Pull request merged**